src/clr-angular/data/datagrid/datagrid-column-toggle-button.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import {Component, Input} from "@angular/core";
+
+import {ColumnToggleButtons, ColumnToggleButtonsService} from "./providers/column-toggle-buttons.service";
+
+@Component({
+    selector: "clr-dg-column-toggle-button",
+    template: `
+        <button
+            (click)="click()"
+            [disabled]="toggleButtons.selectAllDisabled && !isOk()"
+            [ngClass]="getClasses()"
+            type="button">
+            <ng-content></ng-content>
+        </button>
+    `,
+    host: {"[class.action-right]": "isOk()", "[style.display]": "block"}
+})
+
+export class ClrDatagridColumnToggleButton {
+    @Input() type: ColumnToggleButtons;
```

Comment:
Don't we want to name this Input like we do for other ones? `clrToggleButtonType` or something like that?

Reply:
Its not meant to be part of our public API, we don't use the prefix.

Reply:
That said, it is a type, and we don't have a specific naming convention for types (we do for interfaces).

Reply:
I just reread this and realize you meant the input should be `@Input('clrType') type: ColumnToggleButtons;`, fixing.